src/Microsoft.HttpRepl/Commands/BaseHttpCommand.cs

```diff
@@ -205,7 +205,7 @@ private async Task CreateDirectoryStructureForSwaggerEndpointAsync(IShellState s
                     SwaggerUri = programState.SwaggerEndpoint,
                     AllowBaseOverrideBySwagger = false
                 };
-                await apiConnection.SetupHttpState(programState, performAutoDetect: false, cancellationToken).ConfigureAwait(false);
+                await apiConnection.SetupHttpState(programState, performAutoDetect: false, persistHeaders: true, persistPath: true, cancellationToken).ConfigureAwait(false);
```

Comment:
Same question as for `ListCommand` - does it make sense to check the `commandInput` param to see if the user wants to persist headers / path?

Reply:
Same answer as above.